lib/HalModuleParser.js

```diff
@@ -231,81 +231,150 @@ HalModuleParser.prototype = {
 	},
 
 	/*
-	 * tries to determine where we should be looking in the file.
+	 * Validates whether the module binary has a valid
+	 * module prefix at a certain offset.
 	 *
-	 * This changes based on whether or not we're in the bootloader,
-	 * and we attempt to validate the result based on the assumption
-	 * that firmware addresses tend to start with hex '80' since
-	 * they're stored after
-	 * 0x0800xxxx, etc.
+	 * @param {Object} fileBuffer
+	 * @param {number} offset
```

Comment:
Also, when I search for uses of `_validateModulePrefix` I don't actually see this parameter used. So it seems like we're always in non-strict mode?

Reply:
Yeah, the argument not used anywhere, so perhaps we can default to the non-strict behavior and remove it for now entirely. The main problem is that there is no sure-proof way of saying whether what we've found is an actual module prefix or something that just looks like it. This also becomes more problematic if we take into account modules that are not explicitly supported currently in binary-version-reader (e.g. imagine a case where we introduce new platform id or module type, this would cause those checks to fail lowering the score far more than just below max - 1). See a test on `unknown.bin` for an example of that.

There are also other corner cases which I did not want to break as of this PR that may cause some of the checks to fail. For example `RC4_bootloader_pad_BM-09.bin` used in tests will not pass start address - end address == binary size, as the binary is padded.

Reply:
We can also get rid of the score altogether and just output the result of each particular test case that does not immediately fail the validation as a bool in an object and just have the caller look into that?

Reply:
I'm actually really comfortable with the score aspect and get that we need to heuristic it, it's more the boolean pass/fail based on less than 100% that's weird to me. Totally happy with us having a score and choosing best scoring address.

Reply:
Ah, okay, I see what you mean. Yeah, agreed, looks weird.

Let's get rid of it and suffer a hit of having to walk over the full 8KB in case of malformatted or not fully supported binaries (e.g. unknown platform id etc) :+1: ?

Reply:
Fixed.